tools/upgrade/migration/resource/1.4.0/03_server_config_map_upgrade.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# Add the new parameter to server ConfigMap
+# fetch the application.yml and append the new parameter
+APP_YAML="$(oc get configmap syndesis-server-config -o jsonpath='{.data.application\.yml}')
+  integrationStateCheckInterval: 60
```

Comment:
looks a bit fragile, especially if the user should have rearranged the application.yml manually.

I think its ok for now, but we should propably do a more sophisticated configuration handling, like reading in the full yaml in some kind of map (hard with shell, I know, so maybe we provide some simple tools for patching JSON which could be used here).

Then we could read, merge, replace.

But I'm ok for now.

Reply:
I think even having it in Java properties format rather than in YAML would help. I think ideally we would generate migration script (or diff, patch, some kind of input) for the upgrade process when the template is changed. Perhaps in the template generator.

Reply:
@zregvart the integration state check interval is put into bad "section" - it should be in Controllers. And given that it would be hard to put it there, maybe we can leave that one off. 

The default value of 60 should be used even if the property is missing in the yml:
https://github.com/syndesisio/syndesis/blob/528a25f0414bba4adcff326ca8ea3f2b9837913c/app/server/controller/src/main/java/io/syndesis/server/controller/ControllersConfigurationProperties.java#L29

and if the used changed that before, it won't be changed -> profit? :-)